tests/Backend/CommonPart1/DataPreviewLimitsTest.php

```diff
@@ -68,20 +68,65 @@ public function testDataPreviewMaximumLimit()
         }
     }
 
-    /**
-     * @param $rowsCount
-     */
-    private function generateCsv($rowsCount)
+
+    public function testJsonTruncationLimit()
+    {
+        $columnCount = 5;
+        $rowCount = 5;
+        $csvFile = $this->generateCsv($rowCount - 1, $columnCount);
+        $row = [];
+        for ($i = 0; $i < $columnCount; $i++) {
+            $row[] = $this->createRandomString(20000);
+        }
+        $csvFile->writeRow($row);
+        $tableId = $this->_client->createTable($this->getTestBucketId(), 'slim', $csvFile);
+        $jsonPreview = $this->_client->getTableDataPreview($tableId, ['format' => 'json']);
+        $this->assertSame($columnCount, count($jsonPreview['columns']));
+        $this->assertSame($rowCount, count($jsonPreview['rows']));
+        $this->assertContains('col' . ($columnCount - 1), $jsonPreview['columns']);
+        $truncatedRow = $this->getTruncatedRow($jsonPreview);
+        $this->assertNotEmpty($truncatedRow);
+        $this->assertGreaterThan(16000, mb_strlen($truncatedRow[0]['value']), 'Value in row is not truncated');
```

Comment:
Tyhle assery jsou nějaké divné, ne? Prakticky to assertuje, že truncated row je mezi 16000 a 19999 dlouhá. To ale není očekávané chování, ne? Čekal bych, že to bude assertovat, že je to <= 16000 (protože je to truncated). A tím pádem to failne, pokud to bude > 16000...

Reply:
A v historii pull requestu můžeme sledovat jak mě to postupně docházelo ... :-D

Reply:
jaktože jsou všechny buňky v řádku truncated? to info by mělo být pro buňku https://github.com/keboola/connection/issues/1529

Reply:
Pokud je buňka větší než 16383 tak je truncated. Ten test jsem napsal tak, že se všechny buňku v jednom řádku (v tom posledním) vytvoří o délce 20 000. A tím je celý řádek truncated, ale informace isTruncated je vždycky závislá samozřejmě na buňce. Takže výstup vypádá třeba takhle:
```
rows: [
 [
 {
     "columnName": "col1",
     "value": "17vn4avnqqp....strašně moc znaků.....fa5sf5as8df88asdf",
     "isTruncated": 1
 },
 {
     "columnName": "col2",
     "value": "krátky text",
     "isTruncated": 0
 }
 ]
]
```

Reply:
jo super, sorry nepřečetl jsem to celé a jenom mě zarazila tahle věta:) zítra na to mrknou v klidu.
Jenom ještě jede postřeh, "isTruncated" by měl být `bool`.

Reply:
Upravil jsem flag na boolean (testy, connection a UI)

Reply:
Za mě takhle OK 👍